### PR TITLE
Adapt folder layout to enable pandoc integration

### PR DESCRIPTION
**Pull request by vulder**
Enabling automatic document creating with pandoc requires a different folder layout, so we can separate sources files, tools, configs, etc. This PR prepares the repository for further pandoc integration.

TODO:

- [x] Improve spell checking
  - [x] Better spelling error reporting (users want to see file/line numbers) 
  - [ ] Do we need two different spelling exclude lists? 
  - [x] Fix wrong spell checking error
- [x] Test deployment
- [x] Fix hard coded source detection
- [x] Unify multiple places where we lookup/compute the document version
- [x] Regenerate and check tools (old folder)
- [x] Update skeleton to MD table format
- [x] Generate include parts for main.md
- [x] Setup CI for tool unittests

Future TODOs:
- [ ] Refactor preprocessor stage into main generator
  - [ ]  Refactor code
  - [ ] Write tests
- [ ] Refactor building of knowledge_areas section
  - [ ]  Refactor code
  - [ ] Write tests

Pandoc ob ubuntu-20.04 seems to have a bug when rendering one of our topcs. However, misspelling the section works.

### CODE REVIEW

File: sources/introduction.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+is addressed at three proficiency levels.
+These proficiency levels allow each topic to be covered at more than
+one level of detail.
+This allows target audiences with different background and learning
+objectives to be accommodated.
+The three proficiency levels are as follows:
+
+  - foundational.  This level gives the learner the idea that a
+    facility exists, what benefits it offers, and the basic ways of using it.
+    **[Note: Isn't this just "novice"/"beginner"?]**
```
Comment (jcvw) [inline sources/introduction.md:66]: This was determined in https://wg21.link/p2193.  Bjarne was very keen on using the terms foundational, main, advanced.  One thing is that beginner/novece, intermediate suggest that one is better than the other.

File: sources/introduction.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  - main.  This level shows mainstream uses and techniques.
+    For abstraction and organizational mechanisms it also demonstrates how to
+    build them.
+    This level should also give the learner a basic (but not detailed)
+    understanding of how a
+    facility might be implemented so that the learner can have a first-order
+    understanding of any costs involved.
+    **[Note: The term "main" is not very descriptive/helpful.  Could I suggest
```
Comment (jcvw) [inline sources/introduction.md:75]: Same as above, see https://wg21.link/p2193

File: tools/build/make_markdown [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+	if (level == 1) {
+		if (label == "?") {
+			label = bad_char;
+			#basic = emdash;
+			#intermediate = emdash;
+			#advanced = emdash;
+		} else {
+			if (basic == "y") {
+				#basic = sprintf("<a href=\"#%s-basic\">%s</a>", label, check_char);
```
Comment (jcvw) [inline tools/build/make_markdown:20]: You want to keep these commented out lines here?

**Review by jcvw — APPROVED** [review]
Wow, that was a ton of work.  I have a few minor comments, nothing blocking.

File: tools/build/make_markdown (hunk 3 above)
Comment (vulder) [reply tools/build/make_markdown]: Fixed

File: sources/introduction.md (hunk 2 above)
Comment (vulder) [reply sources/introduction.md]: Fixed

File: sources/introduction.md (hunk 1 above)
Comment (vulder) [reply sources/introduction.md]: Fixed

File: sources/course_examples.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+# Examples of Course Curricula
+
+[**NOTE**: Anyone have any suggestions of items to add here?]
+The following are examples of curricula for course on C++:
+...
```
Comment (vulder) [inline sources/course_examples.md:5]: @jcvw What should we do with this section? Keep it, and leave empty with node for future work, or remove?

File: sources/disclaimer.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Disclaimer
+
+This document is intended as a proof of concept to solicit feedback
+from others.
+This document is incomplete.
+This document likely has at least a few errors.
```
Comment (vulder) [inline sources/disclaimer.md:6]: @jcvw Should we keep this?

File: sources/course_examples.md (hunk 4 above)
Comment (jcvw) [reply sources/course_examples.md:5]: I think we can remove this.

File: sources/disclaimer.md (hunk 5 above)
Comment (jcvw) [reply sources/disclaimer.md:6]: I think this can be removed.

**Pull request merged**